src/lxc/start.c

```diff
@@ -1567,7 +1554,8 @@ static int lxc_spawn(struct lxc_handler *handler)
 	 * again.
 	 */
 	if (wants_to_map_ids) {
-		if (!handler->conf->ns_share[LXC_NS_USER]) {
+		if (!handler->conf->ns_share[LXC_NS_USER] ||
+		    (handler->conf->ns_keep & CLONE_NEWUSER) > 0) {
```

Comment:
Shouldn't this be

if (!handler->conf->ns_share[LXC_NS_USER] &&
     (handler->conf->ns_keep & CLONE_NEWUSER) == 0) {
?

Reply:
Could catch, thanks!

Reply:
s/Could/Good/